SPECS/el9/caddy.spec

```
@@ -0,0 +1,189 @@
+%global caddy_config %{_sysconfdir}/caddy
+%global caddy_home %{_sharedstatedir}/caddy
+%global caddy_run %{_rundir}/caddy
+%global caddy_user caddy
+%global caddy_group %{caddy_user}
+%global caddy_uid 517
+%global caddy_gid %{caddy_uid}
+
+Name:           caddy
+Version:        %{rpmbuild_version}
+Release:        %{rpmbuild_release}%{?dist}
+Summary:        Fast and extensible multi-platform HTTP/1-2-3 web server with automatic HTTPS
+License:        ASL 2.0 and MIT and BSD
+URL:            https://github.com/caddyserver/caddy
+Source0:        https://github.com/caddyserver/caddy/archive/v%{version}/caddy-%{version}.tar.gz
+Source1:        caddy-no-binary-mods.patch
+
+BuildRequires:  git
+BuildRequires:  systemd-rpm-macros
+
+
+%description
+Caddy is a powerful, extensible platform to serve your sites, services, and apps, written in Go.
+
+
+%prep
+%autosetup
+cd "${HOME}"
+%{__rm} -fr %{_builddir}/caddy-%{version}
+%{__git} clone --single-branch -b v%{version} %{url} %{_builddir}/caddy-%{version}
+cd %{_builddir}/caddy-%{version}
+%{__patch} -p1 < %{SOURCE1}
```

Why not just use `Patch` instead of `Source`?

```suggestion
Patch0:         caddy-no-binary-mods.patch

BuildRequires:  git
BuildRequires:  systemd-rpm-macros


%description
Caddy is a powerful, extensible platform to serve your sites, services, and apps, written in Go.


%prep
%autosetup -N
cd "${HOME}"
%{__rm} -fr %{_builddir}/caddy-%{version}
%{__git} clone --single-branch -b v%{version} %{url} %{_builddir}/caddy-%{version}
cd %{_builddir}/caddy-%{version}
%autopatch -p1
```

---

Due to a lot of Go build system's reliance on Git project metadata can't be done (easily) without a source checkout.  However, things get wonky in other steps if `%setup` or `%autosetup` macros aren't performed.  Thus, I do a "fiction" where I download the source archive, use `autosetup` to extract it, then immediately delete and clone from GitHub with `git` to the expected location of `rpmbuild`.  If I use `Patch` then autosetup would apply it before being removed.  Hence this lovely alternative hack.

---

Yeah, I saw that "wonky-ness", that's why I added the `-N` option for the `%autosetup` macro in the suggestion.